raft/src/main/java/org/apache/kafka/raft/LeaderAndEpoch.java

```diff
@@ -19,47 +19,15 @@
 import java.util.Objects;
 import java.util.OptionalInt;
 
-public class LeaderAndEpoch {
-    private final OptionalInt leaderId;
-    private final int epoch;
+public record LeaderAndEpoch(OptionalInt leaderId, int epoch) {
     public static final LeaderAndEpoch UNKNOWN = new LeaderAndEpoch(OptionalInt.empty(), 0);
 
     public LeaderAndEpoch(OptionalInt leaderId, int epoch) {
```

Comment:
this could be simplified.
```java
    public LeaderAndEpoch {
        Objects.requireNonNull(leaderId);
    }
```

Also, could you please add unit test for it?